src/lower/merge_lattice.cpp

```diff
@@ -541,11 +541,53 @@ class MergeLatticeBuilder : public IndexNotationVisitorStrict, public IterationA
     bool locateLeft = locateFromLeft(left, right);
 
     // Append all combinations of a and b merge points
-    for (auto& leftPoint : left.points()) {
-      for (auto& rightPoint : right.points()) {
-        points.push_back(intersectPoints(leftPoint, rightPoint, locateLeft));
+    struct pointSort {
+      bool operator()(const MergePoint& a, const MergePoint& b) {
+        size_t left_size  = a.iterators().size() + a.locators().size();
+        size_t right_size = b.iterators().size() + b.locators().size();
+        return left_size > right_size;
+      }
+    } pointSorter;
+
+    // Append all combinations of the merge points of a and b
+    auto sorted_apoint = left.points();
+    auto sorted_bpoint = right.points();
+    std::sort(sorted_apoint.begin(), sorted_apoint.end(), pointSorter);
```

Comment:
Aren't the lattice points already sorted? Why do we need to sort again here?

Reply:
With my new algorithmic changes there were differences in the ordering of points causing `tests-merge_latticce.cpp` to fail in two tests even though the lattices were the same. It permuted some of the points on the same-level. This was the easiest way for me to fix it, but I'm also open to other suggestions.

Reply:
Oh I remember dealing with that. Seems like a good solution to me.